ecl/hql/hqlexpr.cpp

```diff
@@ -914,14 +914,20 @@ void HqlParseContext::setGatherMeta(const MetaOptions & options)
 }
 
 
-static void setDefinitionText(IPropertyTree * target, const char * prop, IFileContents * contents)
+void HqlParseContext::setDefinitionText(IPropertyTree * target, const char * prop, IFileContents * contents)
 {
     StringBuffer sillyTempBuffer;
     getFileContentText(sillyTempBuffer, contents);  // We can't rely on IFileContents->getText() being null terminated..
     target->setProp(prop, sillyTempBuffer);
 
     ISourcePath * sourcePath = contents->querySourcePath();
     target->setProp("@sourcePath", str(sourcePath));
+    if (checkDirty && contents->isDirty())
+    {
+        target->setPropBool("@dirty", true);
+        Owned<IError> error = createError(CategoryInformation, SeverityInformation, WRN_DEFINITION_SANDBOXED, "Definition has been modified", str(contents->querySourcePath()), 0, 0, 0);
```

Comment:
I tried other places - but they ended up calling git status far too often. Putting it here I know that only files that end up in the archive get checked